sds/src/scanner/shared_pool.rs

```diff
@@ -1,21 +1,74 @@
+use std::ops::DerefMut;
+
 use regex_automata::util::pool::{Pool, PoolGuard};
 
 type CachePoolFn<T> = Box<dyn Fn() -> T + Send + Sync>;
-pub type SharedPoolGuard<'a, T> = PoolGuard<'a, T, CachePoolFn<T>>;
+pub type SharedPoolGuard<'a, T, const MAX_POOL_STACKS: usize> =
+    PoolGuard<'a, T, CachePoolFn<T>, MAX_POOL_STACKS>;
 
 /// This is a simple generic wrapper around `Pool` to make it a bit easier to use
 pub struct SharedPool<T> {
-    pool: Pool<T, CachePoolFn<T>>,
+    pool: Box<dyn MyPoolTrait<T>>,
+}
+
+//MyPoolTrait and MyPoolGuardTrait are used to hide away the constant generic in the Pool
+pub trait MyPoolTrait<T>: Sync + Send {
```

Comment:
🫣 I thought these were placeholder names for the traits. We can probably think of something a bit more descriptive (although naming things _is_ the hardest part of programming)

Reply:
🙈🙈🙈🙈
I did see this issue - perhaps I wasn't feeling very inspired last Friday night. 
Anyways I've fixed it now :p